pkg/shelfice/shelfice_init_fixed.F

```
@@ -279,6 +286,31 @@ SUBROUTINE SHELFICE_INIT_FIXED( myThid )
      I      diagName, diagCode, diagUnits, diagTitle, 0, myThid )
 #endif
 
+# ifdef ALLOW_AUTODIFF
+#  ifndef SHI_ALLOW_GAMMAFRICT
```

This will throw an error for the wrong combination of CPP-flags and entries in data.diagnostics. Is that intended? Alternatively one could define all three diagnostics and then some are filled with zeros if specified with the wrong CPP flag.

---

I do prefer to throw an error rather than passively fill zeros, so the user doesn't get the false satisfaction of a completed run all to find the field is just zero... That said the error raised here might be cryptic, and I could add a check in `shelfice_check.F`

---

Ah I see that the forward model does the behavior you're mentioning: passive failure by filling 0's rather than raise an error. However, I would be in favor of putting both the forward and adjoint within the CPP macros and to raise an error rather than fill with 0's. This seems to be the behavior in other packages, e.g. in gmred_diagnostics_init, mom_diagnostics_init